### ISSUE TITLE
Shall I send a PR?

### ISSUE THREAD

**Issue by juek** [florincatalin/AddScriptPluginTypesetter#1]
Hi Florin, 

as you might have already seen, I [forked your plugin](https://github.com/juek/AddScriptPluginTypesetter) and added [quite some stuff](https://github.com/juek/AddScriptPluginTypesetter/commit/b943815cc4cb515a18461d0c80fde35c9f67f0d0). Tell my if I shall send a pull request. Unfortunately, splitting it into multiple PR chunks will not make much sense, since almost all changes I made depend on each other.

**Comment by florincatalin** [comment]
Hallo Juek,

Sorry for the delay, I was caught up with a lot of activity last week.
I downloaded your version and obviously it is much better than the one I made. I appreciate the improvements and will test them the next few days, depending on the time I have available.
Obviously, this is not the case for PR!

**Comment by florincatalin** [comment]
Hi Juek,
I have successfully tested a few applications that required jquery and js, even css code. Usually the css code is preferable to be entered in the template itself, in the head. But for the short css code it worked.
What I don't like about codemirror is that sometimes there are strange problems, see also
https://github.com/JedWatson/react-codemirror/issues/91
It's not because of the plugin but because of the codemirror.
I will test other applications to see how the plugin behaves, but so far it has been excellent!
Cheers,
Florin

**Comment by juek** [comment]
Yep, codemirror tends to look strange sometimes, especially when used inside heavily styled context. must be inheriting some line-height or similar stuff. Maybe I find it.

**Comment by juek** [comment]
... and since we're already off-topic ;o)

Shouldn't we add a Admin page for global scripts, similar to the section type but loading on every page? For facebook tracking and google admanager base scripts and alike.

**Comment by florincatalin** [comment]
> Shouldn't we add a Admin page for global scripts, similar to the section type but loading on every page? 

It would be a good idea!
I haven't had time to test other scripts yet, but this plugin promises to be very useful; especially for those who know a little programming and like to experiment.
I am thinking of making a few pages on my site, in which to apply various scripts, including information on how they can be made. I think it would be useful, we could greatly expand the working capabilities of Typesetter - not really to compete with Wordpress but close enough. Theoretically the possibilities are endless! All you need is work and creativity.

> Yep, codemirror tends to look strange sometimes, especially when used inside heavily styled context. must be inheriting some line-height or similar stuff. Maybe I find it.

Certainly, but so far I have not been able to find out where the problem is ... Anyway, it's a good idea to use codemirror.

**Comment by juek** [comment]
> …codemirror tends to look strange…

fixed with [this commit](https://github.com/juek/AddScriptPluginTypesetter/commit/3014b509414d79858b4d8b8b1c986a2c8be37884)

It was a line-height, as expected.

**Comment by florincatalin** [comment]
It seems that the codemirror works better now, but I found another problem: it seems to change the editing of links (there are triangles in the box and a blue line in the link box), which did not appear before!
I created a [demo page](https://florincatalin.000webhostapp.com/AddScript_-_demo) with a script. 
Others will follow, depending on the time available.

**Comment by juek** [comment]
Do you mean these?

![grafik](https://user-images.githubusercontent.com/704283/69892313-7ca1d900-1304-11ea-9551-b55e6723710e.png)

Its whitespace display, we can make it optional.

**Comment by juek** [comment]
![grafik](https://user-images.githubusercontent.com/704283/69892412-80822b00-1305-11ea-9f8e-f419615cce1a.png)

... thats a tabstop

**Comment by juek** [comment]
> I created a demo page with a script.

Nice! However, such things can also be implemented with small custom plugins like 
[TouchSwipeLoader](https://github.com/juek/TouchSwipeLoader).

**Comment by florincatalin** [comment]
> Do you mean these?

No, this is something else, probably the CSS has changed somewhere.
When you try to insert a link somewhere in the text, something appears:

![image](https://user-images.githubusercontent.com/23093563/69911583-262cbb80-1426-11ea-90f4-71aa8e7c0a9c.png)
and
![image](https://user-images.githubusercontent.com/23093563/69911587-2fb62380-1426-11ea-94aa-26508550e887.png)
I have downloaded and will study your TouchSwipeLoader plugin, thank you!
When I have time I will add other demos to my page.

**Comment by juek** [comment]
These artefacts come from the theme - some Bootswatch themes use rather unusual styling to highlight links or input fields (box-shadow and stuff) - those won't get normalized properly by editors like CKE or Codemirror.

**Comment by florincatalin** [comment]
Another [example](https://florincatalin.000webhostapp.com/AddScript_-_demo_-_part_2), similar to the previous one, but with more configuration possibilities.  I have a few more examples to work on.
